include/ofxColour.h

```diff
@@ -0,0 +1,227 @@
+
+A host must set this property on any effect instances where it has negotiated 
+OCIO colour management (kOfxImageEffectPropColourManagementOCIO).
+Use of URIs for built-in configs, such as ocio://default is permitted.
+
+When native colour management is in use, a host must set this
+property to point to the OCIO config used to define strings for the version of
+OFX it was built against. This will allow a plug-in which uses OCIO to work
+directly with native styles.
+*/
+#define kOfxImageEffectPropOCIOConfig "OfxImageEffectPropOCIOConfig"
+
+/** @brief The colourspace used for this clip
+
+   - Type - string X 1
+   - Property Set - clip instance (read/write)
+   - Valid Values - colourspace that is permitted under the style in use
+
+Hosts should set this property to the colourspace of the input clip. Typically 
+it will be set to the working colourspace of the host but could be any valid 
+colourspace.
+
+Plug-ins may set this property on an output clip. Plug-ins which output motion 
+vectors or similar images which should not be colour managed can use the data 
+colourspace which is present in the built-in OCIO configs.
+
+Both host and plug-in should use the value of 
+kOfxImageClipPropPreferredColourspace where reasonable.
+
+Cross-referencing between clips is possible by setting this property to
+"OfxColourspace_<clip>". For example a plug-in may set this property on its
+output clip to "OfxColourspace_Source", telling the host that the colourspace
+of the output matches the input.
+
+If a clip sets OfxImageClipPropIsMask or it only supports
+OfxImageComponentAlpha, colour management is disabled and this property
+must be unset.
+*/
+#define kOfxImageClipPropColourspace "OfxImageClipPropColourspace"
+
+/** @brief The preferred colourspace for this clip
+
+   - Type - string X N
+   - Property Set - clip instance (read only) and ::kOfxImageEffectActionGetClipPreferences action out args property (read/write)
+   - Valid Values - colourspace that is permitted under the style in use. 
+                    For Basic, any colourspace from ofxColourspaceList.h where IsBasic is true.
+                    For Core, any colourspace from ofxColourspaceList.h where IsCore is true.
+                    For Full, any colourspace from ofxColourspaceList.h.
+                    For OCIO, any string acceptable to Config::getColorSpace().
+
+Plug-ins may set this property during kOfxImageEffectActionGetClipPreferences 
+to request images in a colourspace which is convenient for them. The
+property is an ordered set of colourspace identifiers, which may be any of the
+names or aliases supported by the colour management style in use. If plug-ins
+prefer more esoteric colourspaces, they are encouraged to also include basic
+colourspacesas a fallback. For example a colour grading plug-in which supports
+a specific camera and expects a log colourspace might list:
+
+"arri_logc4", "arri_logc3_ei800", "ACEScct", "ofx_log"
```

Comment:
The thing that most image processing algorithms care about is whether the image is either scene-linear or perceptually uniform (i.e. linear vs. video or log). So I would guess that the most common scenario would be for a plug-in to request the host to provide an image in one of those two formats, but that it would not care specifically which scene-linear or which perceptually uniform color space. In other words, if it wants something scene-linear, it probably could not care less if it's in ACEScg vs. linear Rec.709, and so the host should be free to supply whatever scene-linear color space the clip is already in to avoid unnecessary conversions. 

I'm not clear on how you're saying this should work. Let's say it's an OCIO host and an OCIO plug-in. If the plug-in requests scene_linear, it seems like that is going to trigger a conversion to a specific scene-linear space, right? It seems like there is a need for a more generic option.  Is this what ofx_linear is supposed to be?

But then imagine an OCIO host and a native plug-in. In that case, the OCIO host will load the OFX-specific config file to interact with the plug-in. If the plug-in requests ofx_linear, that is going to be resolved to a specific color space (ACEScg). So again, I'm still not clear on how a plug-in is supposed to say "just give me any scene-linear color space that is convenient for you".

Perhaps the ofx_* roles should be removed from the OFX config so that they never get dereferenced to a specific color space. And some more documentation could be added to indicate that the ofx_* color spaces are always intended to communicate that the least possible conversion should be done (as long as it conforms to that family).

Reply:
I've removed the role definitions from the native config to address this, and explained that the basic colourspaces can be mapped to any colourspace that has the correct reference space and encoding. Hopefully this fully covers the use case you're describing, which I agree will be very common.